src/manifest/deployment/index.ts

```diff
@@ -0,0 +1,378 @@
+import { EnvironmentCollection } from '../resource/Environment/types';
+import { Questions } from 'yeoman-generator';
+
+type DeploymentConfigurationEntry = ConfigurationListEntry<
+  DeploymentDefinition
+>;
+
+const strategies: Array<ListChoiceOptions> = [
+  {
+    name: 'Capistrano',
+    value: 'capistrano',
+  },
+  {
+    name: 'Artifact Repository',
+    value: 'artifact',
+  },
+];
+
+/**
+ * A Manifest sub-generator responsible for prompting and configuration of deployments.
+ *
+ * This generator is dependent on both the Environment and Repository sub-generators as well.
+ *
+ * @class Deployment
+ * @extends {SubGenerator}
+ */
+class Deployment extends SubGenerator {
+  private repositories: RepositoryCollection = {};
+  private environments: EnvironmentCollection = {};
+  private deployments: DeploymentCollection = {};
+  private manifest!: Partial<ManifestDefinition>;
+
+  /**
+   * Execute initialization for this generator.
+   *
+   * @memberof Deployment
+   *
+   * @todo Pre-load configuration from the YAML manifest file.
+   */
+  async initializing() {
+    const config = this.config.getAll();
+
+    this.deployments = config.deployments || {};
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public _getResources(): Record<string, ResourceCollection> {
+    return {
+      deployments: this.deployments,
+    };
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public _setResources(resources: Record<string, ResourceCollection>): void {
+    this.repositories = resources.repositories as RepositoryCollection;
+    this.environments = resources.environments as EnvironmentCollection;
+  }
+
+  /**
+   * Propogate the manifest object in to assign values we're responsible for.
+   *
+   * @param {Partial<ManifestDefinition>} manifest
+   * @memberof Deployment
+   */
+  public _setManifest(manifest: Partial<ManifestDefinition>) {
+    this.manifest = manifest;
+  }
+
+  /**
+   * Create a standardized prompt for selecting a configured repository.
+   *
+   * @param {Partial<ListQuestion>} options
+   *   Override options to contextualize the presentation of the prompt.
+   * @returns {ListQuestion}
+   *   The contextualized prompt ready to be used.
+   * @memberof Deployment
+   */
+  _getRepositorySelectionPrompt(options: Partial<ListQuestion>): ListQuestion {
+    const repositoryOptions: string[] = Object.keys(this.repositories);
+
+    // @todo Add default selection support.
+    const prompt: ListQuestion = {
+      type: 'list',
+      name: 'repository',
+      message: 'What repository should be used?',
+      choices: repositoryOptions,
+      ...options,
+    };
+
+    return prompt;
+  }
+
+  /**
+   * Create a standardized prompt for selecting a configured environment.
+   *
+   * @param {Partial<ListQuestion>} options
+   *   Override options to contextualize the presentation of the prompt.
+   * @returns {ListQuestion}
+   *   The contextualized prompt ready to be used.
+   * @memberof Deployment
```

Comment:
TypeScript ignores the type annotations in these comments - the method declaration is the source of truth here.

```suggestion
   * @param options Override options to contextualize the presentation of the prompt.
   * @returns The contextualized prompt ready to be used.
```